metrics_server/handlers.py

```diff
@@ -8,35 +8,50 @@
 
 from rich.console import Console
 from rich.logging import RichHandler
+from rich.theme import Theme
+
+# Define a custom theme for Rich logging
+custom_theme = Theme({
+    "logging.level.success": "green on black",
+    "logging.level.debug": "blue on black",
+    "logging.level.info": "green on black",
+    "logging.level.warning": "yellow",  # Added a new color
+    "logging.level.error": "bold white on red",
+    "logging.level.critical": "bold magenta on black"
+})
```

Comment:
Why is this defined here again? I would think you should be able to define the logging theme in one place and have it apply everywhere. Also having the same code in two places would make this codebase susceptible to being inconsistent if you were to ever change the theme in one place, but forget to update it in the other.